homeassistant/components/fastdotcom/__init__.py

```diff
@@ -0,0 +1,78 @@
+"""
+Support for testing internet speed via Fast.com.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/fastdotcom/
+"""
+
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_UPDATE_INTERVAL
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['fastdotcom==0.0.3']
+
+DOMAIN = 'fastdotcom'
+DATA_UPDATED = '{}_data_updated'.format(DOMAIN)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_MANUAL = 'manual'
+
+DEFAULT_INTERVAL = timedelta(hours=1)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_UPDATE_INTERVAL, default=DEFAULT_INTERVAL):
+            vol.All(
+                cv.time_period, cv.positive_timedelta
+            ),
+        vol.Optional(CONF_MANUAL, default=False): cv.boolean,
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Set up the Fast.com component."""
+    conf = config[DOMAIN]
+    data = hass.data[DOMAIN] = SpeedtestData(
+        hass, conf[CONF_UPDATE_INTERVAL], conf[CONF_MANUAL]
+    )
+
+    def update(call=None):
+        """Service call to manually update the data."""
+        data.update(no_throttle=True)
+
+    hass.services.async_register(DOMAIN, 'speedtest', update)
+
+    hass.async_create_task(
+        async_load_platform(hass, 'sensor', DOMAIN, {}, config)
+    )
+
+    return True
+
+
+class SpeedtestData:
+    """Get the latest data from fast.com."""
+
+    def __init__(self, hass, interval, manual):
+        """Initialize the data object."""
+        self.data = None
+        self._hass = hass
+        if not manual:
+            self.update = Throttle(interval)(self._update)
```

Comment:
The throttle will only limit calls. It doesn't call the function. Now nothing is calling update on the interval.

Reply:
I totally forgot the `async_track_time_interval(self._hass, self.update, interval)` didn't i?

Thanks for double checking this one...

Reply:
Yes, so we don't need a throttle but just call the function on interval.

Reply:
Is there any value to actually keeping the throttle? Nothing else would be calling update, so I'm thinking it's a little superfluous.